Source/Mosa.DeviceSystem/Bitmap.cs

```diff
@@ -20,16 +20,14 @@ public static Image CreateImage(byte[] data)
 			var height = stream.Read32(0x16);
 			var dataSectionOffset = stream.Read32(0xA);
 
-			var image = new Image(width, height);
+			var image = new Image(width, height, bpp);
 
-			int x = 0;
-			int y = height - 1;
+			uint x = 0;
+			uint y = height - 1;
 
-			for (var i = dataSectionOffset; i < dataSectionOffset + (width * height * bpp); i += bpp)
+			for (var i = dataSectionOffset; i < dataSectionOffset + image.Pixels.Size; i += bpp)
```

Comment:
This has issues for 24bit bitmaps as the standard says that rows of pixels are padded at the end to round up so they are 4byte aligned.
I would recommend changing this loop to handle data section offsets separately as you'd need to add padding to the calculation.

Reply:
Hmm that's interesting. I'll take a look at that.

Reply:
Well, technically, that wouldn't change when rendering, right? Since it's padding at the end but those will pad empty pixels, so it wouldn't change much.

Reply:
It will change the rendering because you're loading empty pad data into the pixels so the image is being corrupted. To load the pixel data correctly you need to skip over the padding.

Reply:
I mean the code right now with skipping the padding. It won't load the empty pad data so it won't corrupt.

Reply:
No, the code right now isn't skipping the padding, the code is assuming that each row of pixels is packed right next to each other continously which is incorrect.

Reply:
You're right. I'll take a look at that.